Keypad/Keypad.js

```diff
@@ -161,18 +162,28 @@ const KeypadBase = kind({
 		return (
 			<Layout {...rest} align="center end" className={css.keypad} inline wrap>
 				{KEY_LIST.map((keyText, rowIndex) => {
+					const {icon, text} = keyText;
+					const isIcon = icon === 'arrowleftturn' || icon === 'phone';
+
+					let ariaLabel = text;
+					if (icon === 'arrowleftturn') {
+						ariaLabel = $L('backspace');
+					} else if (icon === 'phone') {
+						ariaLabel = $L('call');
+					}
+
 					return (
 						<Cell
-							aria-label={keyText.text === 'arrowleftturn' ? $L('Back Space') : keyText.text}
+							aria-label={ariaLabel}
 							component={Key}
 							disabled={disabled}
-							key={`key${rowIndex}-${keyText.text}`}
-							onKeyButtonClick={() => handleInputValue(keyText.icon === 'arrowleftturn' || keyText.icon === 'phone' ? keyText.icon : keyText.text)}
+							key={`key${rowIndex}-${text}`}
+							onKeyButtonClick={() => handleInputValue(isIcon ? icon : text)}
 							shrink
 							label={keyText.label}
-							text={keyText.icon === 'arrowleftturn' || keyText.icon === 'phone' ? null : keyText.text}
+							text={isIcon ? null : text}
 						>
-							{keyText.icon === 'arrowleftturn' || keyText.icon === 'phone' ? keyText.icon : null}
+							{isIcon ? icon : null}
 						</Cell>
 					);
```

Comment:
How about extracting some code in render() out of the kind? In my opinion, it would be enough to execute those code once. By doing it, we could simplify the `render` function.

```suggestion
const keyProps = KEY_LIST.map((keyText, rowIndex) => {
    const {icon, text} = keyText;
    const isIcon = icon === 'arrowleftturn' || icon === 'phone';

    let ariaLabel = text;
    if (icon === 'arrowleftturn') {
        ariaLabel = $L('backspace');
    } else if (icon === 'phone') {
        ariaLabel = $L('call');
    }

    return (
        aria-label={ariaLabel}
        component={Key}
        icon={isIcon ? icon : null}
        key={`key${rowIndex}-${text}`}
        shrink
        label={keyText.label}
        text={isIcon ? null : text}
    );
})

    render: ({disabled, handleInputValue, ...rest}) => {
        return (
            <Layout {...rest} align="center end" className={css.keypad} inline wrap>
                {keyProps.map(({icon, ...rest}) => {
                    return (
                        <Cell
                            {...props}
                            disabled={disabled}
                            onKeyButtonClick={() => handleInputValue(icon || text)}
                        >
                            {icon || null}
                        </Cell>
                    );
                }
            </Layout>
        );
    }
```

Reply:
I don't want to move this code from the render function. It lowers the readability.